modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/model/ApplicationModel.java

```diff
@@ -0,0 +1,370 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.config.spring.dsl.model;
+
+import static org.mule.runtime.config.spring.dsl.processor.xml.CoreXmlNamespaceInfoProvider.CORE_NAMESPACE_NAME;
+import static org.mule.runtime.config.spring.dsl.processor.xml.XmlCustomAttributeHandler.from;
+import static org.mule.runtime.config.spring.dsl.processor.xml.XmlCustomAttributeHandler.to;
+import static org.mule.runtime.core.config.i18n.MessageFactory.createStaticMessage;
+import org.mule.runtime.config.spring.dsl.processor.ApplicationConfig;
+import org.mule.runtime.config.spring.dsl.processor.ConfigFile;
+import org.mule.runtime.config.spring.dsl.processor.ConfigLine;
+import org.mule.runtime.config.spring.dsl.processor.SimpleConfigAttribute;
+import org.mule.runtime.core.api.MuleRuntimeException;
+import org.mule.runtime.core.api.config.ConfigurationException;
+
+import com.google.common.collect.ImmutableSet;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+
+/**
+ * An {@code ApplicationModel} holds a representation of all the artifact configuration using an abstract model
+ * to represent any configuration option.
+ *
+ * This model is represented by a set of {@link org.mule.runtime.config.spring.dsl.model.ComponentModel}. Each {@code ComponentModel}
+ * holds a piece of configuration and may have children {@code ComponentModel}s as defined in the artifact configuration.
+ *
+ * Once the set of {@code ComponentModel} gets created from the application {@link org.mule.runtime.config.spring.dsl.processor.ConfigFile}s
+ * the {@code ApplicationModel} executes a set of common validations dictated by the configuration semantics.
+ *
+ * @since 4.0
+ */
+public class ApplicationModel
+{
+
+    public static final String MULE_ROOT_ELEMENT = "mule";
```

Comment:
add a TODO for moving this constants and their underlying rules to someplace else. ApplicationModel should just model the application, not the language rules since we might end up having several languages